product_attribute_model_link/migrations/16.0.2.0.0/pre-migrate.py

```diff
@@ -0,0 +1,33 @@
+# Copyright (C) 2024 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+def migrate(cr, version):
+    # Add new columns (model and res_id) to the table
+    cr.execute(
+        """
+        ALTER TABLE product_attribute_value
+        ADD COLUMN IF NOT EXISTS model VARCHAR,
+        ADD COLUMN IF NOT EXISTS res_id INTEGER
+    """
+    )
+    # Extract and store model name and res ID from the linked_record_ref field
+    cr.execute(
+        """
+        SELECT id, linked_record_ref
+        FROM product_attribute_value
+        WHERE linked_record_ref IS NOT NULL
+    """
+    )
+    for rec_id, linked_ref in cr.fetchall():
```

Comment:
Consider doing
```sql
UPDATE product_attribute_value SET
    model = split_part(linked_record_ref, ',', 1), 
    res_id = split_part(linked_record_ref, ',', 2)
```

Reply:
Thanks! Corrected. Could you check again?